rspace/src/main/scala/coop/rchain/rspace/history/LMDBTrieStore.scala

```diff
@@ -110,20 +95,14 @@ class LMDBTrieStore[K, V] private (
     getRoot(txn, branch)
       .find(_ == hash)
       .orElse {
-        getPastRootsInBranch(txn, branch)
-          .find(_ == hash)
-          .map { blake: Blake2b256Hash =>
-            putRoot(txn, branch, blake)
-            blake
-          }
-      }
-      .orElse {
-        getAllPastRoots(txn)
-          .find(_ == hash)
-          .map { blake: Blake2b256Hash =>
-            putRoot(txn, branch, blake)
-            blake
-          }
+        val hashBytes  = Codec[Blake2b256Hash].encode(hash).map(_.bytes.toDirectByteBuffer).get
+        val maybeValue = Option(_dbPastRoots.get(txn, hashBytes))
+        maybeValue
+          .map(_ => hash)
+          .map(hash => {
+            putRoot(txn, branch, hash)
```

Comment:
Double check the ordering here because other ones are mostly

function(..., hash, branch)

in that order.

Reply:
I guess the Branch vs Blake2b256Hash type would catch it?

Reply:
yep